File: demo/test/temperature-conversion.test.js
import { expect } from "chai";
import TemperatureConversion from "../src/temperature-conversion.js";

describe("TemperatureConversion", () => {
  const tempConv = new TemperatureConversion();

  it("should convert Celsius to Fahrenheit correctly", () => {
    expect(tempConv.celsiusToFahrenheit(0)).to.equal(32);
    expect(tempConv.celsiusToFahrenheit(100)).to.equal(212);
  });

  it("should convert Celsius to Kelvin correctly", () => {
    expect(tempConv.celsiusToKelvin(0)).to.equal(273);
    expect(tempConv.celsiusToKelvin(100)).to.equal(373);
  });

  it("should convert Celsius to Rankine correctly", () => {
    expect(tempConv.celsiusToRankine(0)).to.equal(492);
    expect(tempConv.celsiusToRankine(100)).to.equal(672);
  });

  it("should convert Fahrenheit to Celsius correctly", () => {
    expect(tempConv.fahrenheitToCelsius(32)).to.equal(0);
    expect(tempConv.fahrenheitToCelsius(212)).to.equal(100);
  });

  it("should convert Fahrenheit to Kelvin correctly", () => {
    expect(tempConv.fahrenheitToKelvin(32)).to.equal(273);
    expect(tempConv.fahrenheitToKelvin(212)).to.equal(373);
  });

  it("should convert Kelvin to Celsius correctly", () => {
    expect(tempConv.kelvinToCelsius(273)).to.equal(0);
    expect(tempConv.kelvinToCelsius(373)).to.equal(100);
  });

  it("should convert Kelvin to Fahrenheit correctly", () => {
    expect(tempConv.kelvinToFahrenheit(273)).to.equal(32);
    expect(tempConv.kelvinToFahrenheit(373)).to.equal(212);
  });
});

File: python/src/temperature_conversion_test.py
import unittest
from temperature_conversion import TemperatureConversion

class TestTemperatureConversion(unittest.TestCase):
    def setUp(self):
        self.temp_conv = TemperatureConversion()

    def test_celsius_to_fahrenheit(self):
        self.assertEqual(self.temp_conv.celsius_to_fahrenheit(0), 32)
        self.assertEqual(self.temp_conv.celsius_to_fahrenheit(100), 212)

    def test_celsius_to_kelvin(self):
        self.assertEqual(self.temp_conv.celsius_to_kelvin(0), 273)
        self.assertEqual(self.temp_conv.celsius_to_kelvin(100), 373)

    def test_celsius_to_rankine(self):
        self.assertEqual(self.temp_conv.celsius_to_rankine(0), 492)
        self.assertEqual(self.temp_conv.celsius_to_rankine(100), 672)

    def test_fahrenheit_to_celsius(self):
        self.assertEqual(self.temp_conv.fahrenheit_to_celsius(32), 0)
        self.assertEqual(self.temp_conv.fahrenheit_to_celsius(212), 100)

    def test_fahrenheit_to_kelvin(self):
        self.assertEqual(self.temp_conv.fahrenheit_to_kelvin(32), 273)
        self.assertEqual(self.temp_conv.fahrenheit_to_kelvin(212), 373)

    def test_kelvin_to_celsius(self):
        self.assertEqual(self.temp_conv.kelvin_to_celsius(273), 0)
        self.assertEqual(self.temp_conv.kelvin_to_celsius(373), 100)

    def test_kelvin_to_fahrenheit(self):
        self.assertEqual(self.temp_conv.kelvin_to_fahrenheit(273), 32)
        self.assertEqual(self.temp_conv.kelvin_to_fahrenheit(373), 212)

if __name__ == '__main__':
    unittest.main()
